operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -240,7 +241,19 @@ func CreateOrUpdateLokiStack(
 		}
 	}
 
-	objects, err := manifests.BuildAll(opts)
+	var apiServer openshiftv1.APIServer
+	if err = k.Get(ctx, client.ObjectKey{Name: "cluster"}, &apiServer); err != nil {
+		if apierrors.IsNotFound(err) {
+			return &status.DegradedError{
+				Message: "Missing apiserver object",
+				Reason:  lokiv1.ReasonMissingOpenShiftAPIServer,
+				Requeue: false,
+			}
+		}
+		return kverrors.Wrap(err, "failed to lookup apiServer")
+	}
+
+	objects, err := manifests.BuildAll(opts, apiServer.Spec.TLSSecurityProfile)
```

Comment:
I wonder if the better option here is to expand the `Options` structure to also have an optional TLS configuration pieces from the `apiServer` object. 

Also, returning an error here causes an issue in base K8s, no? We should only return a degraded error if we are on an OpenShift cluster.

Reply:
I'd agree that adding it to `Options` is probably a better deal. We can also convert the `TLSSecurityProfile` to the information we actually need (minimum TLS version and cipher list) before passing it into the manifest generation.

The request to get the APIServer resource can only be done on OpenShift and will fail on other clusters. But even on OpenShift it can not be assumed that the TLSSecurityProfile is configured. This leads to the question what a sensible fallback is, I'd say either we go with the intermediate profile like the PR implementation already does or we skip configuring the TLS version and ciphers like we did before.